dwave/optimization/model.py

```diff
@@ -412,7 +412,15 @@ def to_file(self, **kwargs) -> typing.BinaryIO:
             :meth:`.into_file`, :meth:`.from_file`
         """
         file = tempfile.TemporaryFile(mode="w+b")
-        self.into_file(file, **kwargs)
+
+        # into_file can raise an exception, in which case we close off the
+        # tempfile before returning
+        try:
+            self.into_file(file, **kwargs)
+        except Exception as exception:
+            file.close()
+            raise exception
```

Comment:
How about:
```suggestion
        try:
            self.into_file(file, **kwargs)
        finally:
            file.close()
```

Reply:
actually, I see the file is used below.. so ignore this :)

Reply:
but could at least simplify like;
```suggestion
        try:
            self.into_file(file, **kwargs)
        except:
            file.close()
            raise
```

Reply:
Ah! TIL. Though I feel like that syntax is a violation of the explicit is better than implicit.

Reply:
Also interesting: https://peps.python.org/pep-0760/

Reply:
Yeah, there's an argument that can be made you shouldn't catch `BaseException`, so could do:
```
except Exception:
    raise
```

But I don't completely agree that bare except is implicit here. It's pretty explicit you want to close the file on _any_ error and re-raise it then. (The use case over which that PEP was rejected, I guess)